modules/market/src/main/java/com/opengamma/strata/market/curve/CurveMetadata.java

```diff
@@ -65,6 +65,15 @@
   public abstract Optional<DayCount> getDayCount();
 
   /**
+   * Gets the calibration information, optional.
+   * <p>
+   * When a curve is calibrated, useful information is obtained, which can be stored here.
+   * 
+   * @return the calibration information
+   */
+  public abstract Optional<CurveCalibrationInfo> getCalibrationInfo();
```

Comment:
It is all pretty important stuff now, not just the calibration info. The x/y types must be set, as must the day count and name. There is potential to rename `CurveMetadata` to `CurveInfo`,